File: examples/vulnerable_app/service.go
package main

import (
	"crypto/md5"
	"crypto/tls"
	"database/sql"
	"fmt"
	"os/exec"
)

func run(input string) {
	exec.Command("sh", "-c", input).Run()
}

func query(db *sql.DB, id string) {
	db.Query(fmt.Sprintf("SELECT * FROM users WHERE id=%s", id))
}

func hash(data []byte) {
	_ = md5.New()
}

func client() *tls.Config {
	return &tls.Config{InsecureSkipVerify: true}
}

File: examples/vulnerable_app/Service.java
import java.io.ObjectInputStream;
import java.security.MessageDigest;
import java.sql.Statement;

public class Service {
    void run(String cmd) throws Exception {
        Runtime.getRuntime().exec(cmd);
    }

    void query(Statement stmt, String id) throws Exception {
        stmt.executeQuery("SELECT * FROM users WHERE id=" + id);
    }

    void hash() throws Exception {
        MessageDigest.getInstance("MD5");
    }

    Object load(ObjectInputStream in) throws Exception {
        return in.readObject();
    }
}
